src/test/java/com/cegeka/academy/repository/InterestTest.java

```diff
@@ -61,4 +67,58 @@ public void testFindByDescription(){
         assertThat(interestRepository.findByDescription("description1").get(0)).isEqualTo(interestRepository.findAll().get(0));
     }
 
+    @Test
+    public void testAddUserInterest(){
+        Interest interest1  = new Interest();
+        interest1.setName("interest1");
+        interest1.setDescription("description1");
+
+        User user = new User();
+        user.setEmail("gigi.gogaie@gmail.com");
+        user.setFirstName("Gogaie");
+        user.setLastName("Momaie");
+        user.setPassword("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        user.setLogin("gigel@purcel.com");
+
+        Set userSet = new HashSet<>();
+        Set interestSet = new HashSet<>();
+
+        userSet.add(user);
+        interestSet.add(interest1);
+
+        user.setUserInterests(interestSet);
+        interest1.setInterestUsers(userSet);
+
+        interestRepository.save(interest1);
+        userRepository.save(user);
+
+        assertThat(interestRepository.findAll().size()).isEqualTo(1);
+        assertThat(interestRepository.findAll().get(0).getName()).isEqualTo("interest1");
+        assertThat(interestRepository.findAll().get(0).getDescription()).isEqualTo("description1");
+        for(int i =0 ;i < userRepository.findAll().size(); i++)
+        {
+            System.out.println(i + ": "+ userRepository.findAll().get(i).toString());
+        }
+
+        assertThat(userRepository.findAll().get(4).getUserInterests().equals(interestSet)).isTrue();
+
+        for(Object o : userRepository.findAll().get(4).getUserInterests().toArray())
```

Comment:
`userRepository.findAll()` ar trebuie sa intoarca un obiect de tip `User` iar `user.getUserInterests()` o lista de `UserInterests` deci nu e nevoie de `toArray` ca sa iterezi prin lista si nici de `Object o` ca sa obtii fiecare element. Eu as scrie codul astfel `for(UserInterest ui : userRepository.findAll().get(4).getUserInterests()) {...}`

Reply:
@binacatalin, thanks, I didn't knew we can iterate over a Set like that. So what shall I do? Shall I keep the Syso's or just the asserts?

Reply:
Doar assert-urile. ;)